### ISSUE TITLE
Code coverage

### ISSUE THREAD

**Issue by tmpfs** [cgiffard/node-simplecrawler#225]
Are you guys interested in a PR that adds code coverage? I use `istanbul` and find it is excellent.

This would be a non-breaking change that would be beneficial for everybody.

**Comment by cgiffard** [comment]
I used `istanbul` for my library BAS, and worked my way up to a high level of coverage. It was certainly a nice tool to use. Ultimately though, I think the methodology can have problems, and I'd prefer behaviour completeness to code coverage. (Every new feature added should have tests for every behaviour.)

I actually wrote a rough PoC tool to solve some problems with the code coverage methodology, and you can read the rationale for it over here. https://github.com/cgiffard/SteamShovel

Our immediate problem is that Simplecrawler's tests are more like integrations and are not heavily isolated. So I fear we won't get much visibility from traditional code coverage.

It wouldn't *hurt*, but it's not a priority for me. :)

**Comment by tmpfs** [comment]
I agree with you that asserting on behaviour is the only true test of correctness, however in the past I have found that coverage has helped me to find areas that need testing and sometimes subtle bugs.

It can also help to find behaviours that have yet to be tested - especially error handling behaviour (which is often very difficult to test).

Steamshovel looks very interesting, unfortunately I don't have the time to look closely at it right now, parked for future reference ;)

Closing as not a priority.
